winterfell/webapp/src/js/claimSelectFormsCtrl.js

```diff
@@ -1,55 +1,17 @@
 var app = angular.module('vetafiApp');
-app.controller('claimSelectFormsCtrl', ['$scope', '$state', '$stateParams', 'claimService', 'formTemplateService',
-  function($scope, $state, $stateParams, claimService, formTemplateService) {
+app.controller('claimSelectFormsCtrl', ['$scope', 'claimService', 'formTemplateService', '$stateParams', 'forms',
+  function($scope, claimService, formTemplateService, $stateParams, forms) {
     $scope.claimId = $stateParams.claimId;
+    $scope.formTemplates = formTemplateService;
+    $scope.forms = _.keyBy(forms, function(form) {
+      return form.key;
+    });
 
-    $scope.allForms = formTemplateService; // returns an object {formId -> {...}}
-    var userForms = {};                    // should be {formId -> {...}}
-    // TODO: state of all forms edited by user
-    // var userForms = {
-    //   'VBA-21-0966-ARE': {
-    //      state: 'complete'
-    //   }
-    // };
-    $scope.requiredForms = [];
-    $scope.optionalForms = [];
-    $scope.numRequiredCompleted = 0;
-    $scope.progressBarType = undefined;
+    $scope.numRequiredCompleted = _.sum(_.map(forms, function (form) {
+      return form.answered == form.answerable ? 1 : 0;
+    }));
 
-    function partitionAllForms() {
-      var partitionForms = _.partition($scope.allForms, 'required');
-      $scope.requiredForms = partitionForms[0];
-      $scope.optionalForms = partitionForms[1];
-    }
-
-    function calcFormState(state, required) {
-      if (state == 'complete') {
-        return 'complete';
-      } else if (required) {
-        return 'required_incomplete';
-      } else {
-        return 'incomplete';
-      }
-    }
-
-    function calcFormsCompleted() {
-      var numRequiredCompleted = 0;
-
-      _.map(_.keys($scope.allForms), function(formId) {
-        var targetForm = $scope.allForms[formId];
-        var userForm = userForms[formId];
-        var userFormState = userForm ? userForm.state : 'incomplete';
-        targetForm.id = formId;
-        targetForm.state = calcFormState(userFormState, targetForm.required);
-
-        if (targetForm.required && targetForm.state == 'complete') {
-          numRequiredCompleted++;
-        }
-      });
-
-      $scope.numRequiredCompleted = numRequiredCompleted;
-      $scope.progressBarType = $scope.numRequiredCompleted == $scope.requiredForms.length ? 'success' : undefined;
```

Comment:
`$scope.progressBarType` is missing. Logic is pretty simple. Could you add this line maybe after the line below?